### ISSUE TITLE
Creating http profile does not accept any input for insertHeader on header name or value.

### ISSUE THREAD

**Issue by clearpilot** [F5Networks/f5-appsvcs-extension#516]
### Environment
 * Application Services Version: 3.31.0-6
 * BIG-IP Version: BIG-IP 15.1.3.1 Build 0.0.18 Point Release 1 

### Summary
Unable to use any value for "insertHeader" when creating an http profile.

This is reproducible every time.  There seem to be no variation of text that the insertHeader will accept.

### Steps To Reproduce
1. Submit the following declaration:
Example of Config needed, this works with tmsh and is a valid configuration.
```
                "pro_http_myfqdn_80": {
                    "class": "HTTP_Profile",
                    "xForwardedFor": true,
                    "insertHeader": "Proxy-Client-Data: [IP::remote_addr]\:[TCP::remote_port]",
                }
```

2. Observe the following error response:
```json
{
    "json": {
        "code": 422,
        "declarationFullId": "",
        "errors": [
            "/as3_temp_build_partition/app_web_appvip/pro_http_myfqdn_80/insertHeader: should be object"
        ],
        "message": "declaration is invalid"
    },
}
```

### Expected Behavior
This should be an accepted header to insert, create new http profile with the insert header.

### Actual Behavior
A clear and concise description of what actually happens.
Please include any applicable error output.
Error 422

#####
Reviewing the code it looks like the regex patters are not as needed for the fields.

Code Snippit of existing....
```json
        "insertHeader": {
          "title": "Insert header",
          "description": "You may insert one header into each request before AS3 sends it to a pool member.  The header value may be a simple string or the result of an iRules TCL expression (for example, [IP::client_addr]).  This is the most efficient way to insert a single header; to insert multiple headers use an iRule or an Endpoint policy",
          "type": "object",
          "properties": {
            "name": {
              "title": "Header name",
              "description": "Name of the HTTP header to insert",
              "type": "string",
              "pattern": "^[^\\x00-\\x20\\x22:\\x5c\\x7f-\\xff]+$",
              "minLength": 1,
              "maxLength": 128
            },
            "value": {
              "title": "Header value",
              "description": "May contain iRules TCL expression",
              "type": "string",
              "pattern": "^[^\\x00-\\x1f\\x7f-\\xff]*$"
            }
          },
          "additionalProperties": false
        },
```

It seems the patters for both need to be updated to something like.... [A-Za-z\[\]][0-9A-Za-z_.-:\\[\]]*$

Hope this helps.

Thanks

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-2786.

**Comment by RobCupples** [comment]
What the error and schema are saying is this needs to be an object.  This should be written as

```
                "pro_http_myfqdn_80": {
                    "class": "HTTP_Profile",
                    "xForwardedFor": true,
                    "insertHeader": {
                        "name": "Proxy-Client-Data",
                        "value": "[IP::remote_addr]\\:[TCP::remote_port]"
                    }
                }
```
Also as you can also see I have escaped the backslash with an additional backslash.  This needs to be done for JSON.

There is still a bug in AS3 however.  This works but is not what we call idempotent.  If you post this twice in a row AS3 will report "success" in the result instead of "no change".

**Comment by dstokesf5** [comment]
This has been resolved in the 3.33.0 release.
